src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -363,11 +363,10 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
   // Note:
   // we take the most significant bits of the hashcode, not the lower ones
   // this is of crucial importance when populating the table in parallel
-  protected final def index(hcode: Int) = {
+  protected final def index(hcode: Int): Int = {
     val ones = table.length - 1
-    val improved = improve(hcode, seedvalue)
-    val shifted = (improved >> (32 - java.lang.Integer.bitCount(ones))) & ones
-    shifted
+    val exponent = Integer.numberOfLeadingZeros(ones)
```

Comment:
* `numberOfLeadingZeros` might be optimized in the future and is already fast enough for now.
* for some reason extracting it as a variable didn't improve the speed at all :/